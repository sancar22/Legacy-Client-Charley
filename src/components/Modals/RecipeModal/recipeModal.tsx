import React from 'react';

import { IRecipe } from 'src/interfaces';

import * as styles from './recipeModal.module.css';

interface RecipeModalProps {
  show: boolean;
  handleClose: () => void;
  recipe: IRecipe;
}

const RecipeModal: React.FC<RecipeModalProps> = ({
  show,
  handleClose,
  recipe,
}): JSX.Element => (
  <div className={show ? styles.modalShow : styles.modalHide}>
    <div className={styles.closeIcon} onClick={handleClose} aria-hidden='true'>
      x
    </div>
    <div className={styles.title}>{recipe.name}</div>

    <div className={styles.content}>
      <div className={styles.recipeYield}>{recipe.recipeYield}</div>
      <ul>
        {recipe.recipeIngredient.map((ingredient, index) => (
          <li className={styles.ingredient} key={index}>
            {ingredient}
          </li>
        ))}
      </ul>
      <div className={styles.prepareHeader}>Prepare</div>
      {recipe.recipeInstructions.map((step, index) => (
        <div key={index}>
          <h4 className={styles.step__header}>STEP{` ${index + 1}`}</h4>
          <p className={styles.step__task}>{step}</p>
        </div>
      ))}
      {recipe.notes.length ? (
        <>
          <div className={styles.prepareHeader}>Notes</div>
          {recipe.notes.map((note, index) => (
            <li key={index} className={styles.text__note}>
              {note.text}
            </li>
          ))}
        </>
      ) : null}
    </div>
  </div>
);

export default RecipeModal;
