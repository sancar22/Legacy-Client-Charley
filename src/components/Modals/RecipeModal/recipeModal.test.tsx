import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import reduxStore from '../../../state/store';
import RecipeModal from './recipeModal';

const { store } = reduxStore();

describe('recipeModal component', () => {
  const recipeModal = create(
    <Provider store={store}>
      <RecipeModal show={true} handleClose={jest.fn()} recipe = {{
        _id: '',
        notes: [],
        origin: '',
        url: '',
        recipeIngredient: [],
        recipeInstructions: [],
      }}/>
    </Provider>,
  );
  test('match previous recipeModal snapshot', () => {
    expect(recipeModal).toMatchSnapshot();
  });
});
