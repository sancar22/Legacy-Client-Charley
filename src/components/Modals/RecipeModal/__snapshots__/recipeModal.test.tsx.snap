// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipeModal component match previous recipeModal snapshot 1`] = `
<div>
  <div
    aria-hidden="true"
    onClick={[MockFunction]}
  >
    x
  </div>
  <div />
  <div>
    <div />
    <ul />
    <div>
      Prepare
    </div>
  </div>
</div>
`;
